### PR TITLE
Use Aria tool for downloading image files

### PR DESCRIPTION
**Pull request by dsrivastavv**

### CODE REVIEW

File: lisa/tools/tar.py [hunk 1]
```diff
@@ -6,13 +6,25 @@
...
 
 class Tar(Tool):
     @property
     def command(self) -> str:
         return "tar"
 
+    def _check_exists(self) -> bool:
+        if (
+            isinstance(self.node.os, Suse)
+            and not self.node.tools[Bzip2]._check_exists()
```
Comment (squirrelsc) [inline lisa/tools/tar.py]: The Bzip2 can be added as a dependency of Tar, if it's required.

Comment (squirrelsc) [reply lisa/tools/tar.py]: Avoid to call private method of other classes. If it happens, it means the best practices are not followed, or the class design should be changed.

File: lisa/tools/bzip2.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from lisa.executable import Tool
+
+
+class Bzip2(Tool):
+    @property
+    def command(self) -> str:
+        return "bzip2"
```
Comment (squirrelsc) [inline lisa/tools/bzip2.py:10]: Can bzip2 be installed on Suse?

File: lisa/tools/aria.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+from pathlib import PurePosixPath
+from typing import TYPE_CHECKING
+
+from lisa.base_tools import Wget
+from lisa.executable import Tool
+from lisa.tools.make import Make
+from lisa.tools.tar import Tar
+
+if TYPE_CHECKING:
+    from lisa.operating_system import Posix
```
Comment (squirrelsc) [inline lisa/tools/aria.py]: the os reference won't bring cycle import for this tool. If so, just import as normal.

File: lisa/tools/bzip2.py (hunk 2 above)
Comment (dsrivastavv) [reply lisa/tools/bzip2.py:10]: yes. I have added the install command.

File: lisa/tools/tar.py (hunk 1 above)
Comment (dsrivastavv) [reply lisa/tools/tar.py]: This requirement is only for Suse. I checked on ubuntu and Redhat and it works fine

Comment (squirrelsc) [reply lisa/tools/tar.py]: When calling the `tools[Bzip2]`, the `_check_exists` is already called, so it's not useful to call `_check_exists`.

File: lisa/tools/aria.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    def get(
+        self,
+        url: str,
+        file_path: str = "",
+        filename: str = "",
+        overwrite: bool = True,
+        sudo: bool = False,
+        force_run: bool = False,
+        num_connections: int = 2,
```
Comment (LiliDeng) [inline lisa/tools/aria.py]: use bigger number than 2 like 10 to speedup download

File: lisa/tools/aria.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        timeout: int = 600,
+    ) -> str:
+
+        if file_path:
+            # create folder when it doesn't exist
+            self.node.shell.mkdir(PurePosixPath(file_path), exist_ok=True)
```
Comment (LiliDeng) [inline lisa/tools/aria.py]: use mkdir tool than this, probably there is no permission when use shell mkdir

File: lisa/tools/aria.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        else:
+            file_path = str(self.node.working_path)
+
+        download_path = f"{file_path}/{filename}"
+        download_pure_path = self.node.get_pure_path(download_path)
+        if overwrite and self.node.shell.exists(download_pure_path):
+            self.node.shell.remove(download_pure_path, recursive=True)
+            force_run = True
```
Comment (LiliDeng) [inline lisa/tools/aria.py]: seems this tool has `--allow-overwrite` this parameter, if so, please use this parameter.

File: lisa/tools/aria.py (hunk 4 above)
Comment (dsrivastavv) [reply lisa/tools/aria.py]: This might be a function of number of cores. Let me set it to number of cores instead

File: lisa/tools/tar.py (hunk 1 above)
Comment (dsrivastavv) [reply lisa/tools/tar.py]: Sure! I will remove this call

File: lisa/tools/aria.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+    def get(
+        self,
+        url: str,
+        file_path: str = "",
+        filename: str = "",
+        overwrite: bool = True,
+        sudo: bool = False,
+        force_run: bool = False,
+        num_connections: Optional[int] = None,
```
Comment (LiliDeng) [inline lisa/tools/aria.py:59]: max-connection-per-server must be between 1 and 16.

File: lisa/tools/aria.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+            self.node.tools[Mkdir].create_directory(file_path, sudo=sudo)
+        else:
+            file_path = str(self.node.working_path)
+
+        # set download path
+        download_path = f"{file_path}/{filename}"
+
+        # if num_connections is not specified, set to the number of cores on the node
+        if not num_connections:
+            num_connections = self.node.tools[Lscpu].get_core_count()
```
Comment (LiliDeng) [inline lisa/tools/aria.py]: max-connection-per-server must be between 1 and 16.
min(self.node.tools[Lscpu].get_core_count(), 16) ?

File: lisa/tools/aria.py (hunk 3 above)
Comment (squirrelsc) [reply lisa/tools/aria.py]: The bzip2 is not changed.

File: lisa/tools/aria.py (hunk 5 above)
Comment (dsrivastavv) [reply lisa/tools/aria.py]: fixed!

File: lisa/tools/aria.py (hunk 8 above)
Comment (dsrivastavv) [reply lisa/tools/aria.py]: fixed!

File: lisa/tools/aria.py [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def install(self) -> bool:
+        posix_os: Posix = self.node.os  # type: ignore
+        try:
+            posix_os.install_packages(["aria2"])
+        except Exception as identifier:
+            self._log.debug(f"Failed to install aria2 from apt: {identifier}")
```
Comment (LiliDeng) [inline lisa/tools/aria.py]: apt is Debian only, right?

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**